offering-profile/validate.ts

```diff
@@ -20,10 +20,137 @@ export async function validateEmail(
   return vlf.lformValidationSuccess;
 }
 
+// validate the company name to which the offering belongs
+export async function validateCompanyName(
+  ctx: vlf.LhcFormValidationContext,
+): Promise<vlf.LhcFormValidResult | vlf.LhcFormInvalidResult> {
+  if (!ctx.form.name) { //expecting ctx.form.items[].value with actual value
```

Comment:
You need to write the actual code here that will perform the check against values in `ctx.form.*` properties and sub-properties. Please do the same in each function. Once I see enough examples I'll generalize your code and put it into lforms package directly.

Reply:
Tried to add example validation rules using `for` loop. When we try the same with the `filter` we faced some type-safe issues

Reply:
Ok, please continue working on the structural rules, I'll accept these and build the infrastructure.